deepeval/metrics/g_eval/g_eval.py

```diff
@@ -284,16 +284,23 @@ def generate_weighted_summed_score(
             # Then, calculate the score based on the logprobs
             token_linear_probability: Dict[int, float] = {}
             sum_linear_probability = 0
+            # Filter out tokens with <1% linear probability, i.e., logprobs < math.log(0.01)
+            min_logprob = math.log(0.01)
             for token_logprob in score_logprobs["top_logprobs"]:
-                # Filter out tokens with <1% linear probability, i.e., logprobs < math.log(0.01)
                 logprob = token_logprob["logprob"]
 
-                if logprob < math.log(0.01):
+                # Filter out low probability tokens
+                if logprob < min_logprob:
+                    continue
+                # Filter out non-decimal token to prevent errors in later int(token) conversion
+                if not token_logprob["token"].isdecimal():
```

Comment:
Why not just use the `logprob` variable?

Reply:
Oh basically the `token_logprob` dict looks like `{'token': '9', 'bytes': [57], 'logprob': -0.18066935},` `logprob` is the log-probability for the given `token`, so we'll need to retain both for calculating `token` * probabillity.